extension/pq/listener.go

```diff
@@ -92,6 +100,7 @@ func (s *Listener) Listen(ctx context.Context, exec sql.ProjectionTrigger) error
 		case n := <-listener.Notify:
 			// Unmarshal the notification
 			notification := s.unmarshalNotification(n)
+			s.metrics.ReceivedNotification(notification != nil)
```

Comment:
You can actually move this after line 100 since `n` is `nil` when for example a reconnect occurs